src/service/resources-impl.js

```diff
@@ -624,6 +625,12 @@ export class ResourcesImpl {
         const ampElements = element.getElementsByClassName('i-amphtml-element');
         for (let i = 0; i < ampElements.length; i++) {
           const r = Resource.forElement(ampElements[i]);
+          userAssert(
+            typeof r !== 'undefined',
+            'AMP Element is missing an associated resource. Element: %s, Mode: %s',
+            ampElements[i],
+            getMode()
```

Comment:
You can't use `getMode()` like this, you have to grab individual properties off of it. It's best to not think of it as an object value.

Reply:
Where does all the special "getMode" handling happen? I'm guessing closure compiler does something magic here like inlining values?

Reply:
It's a babel transform.

Reply:
Ah, I found `build-system/babel-plugins/babel-plugin-amp-mode-transformer/index.js`. That explains "test", "localDev" and "minified", but what about the other properties?

Reply:
They'll be computed as needed.